packages/push/src/PushRegistration.ts

```diff
@@ -96,22 +97,102 @@ export class PushRegistration {
       baseURL: url,
       timeout: 5000,
       headers: {"Authorization": `Basic ${authToken}`}
     });
 
     return new Promise((resolve, reject) => {
-      return instance.post(PushRegistration.API_PATH, postData)
+      return http.post(PushRegistration.API_PATH, postData)
       .then(
         () => {
+          const storage = window.localStorage;
+          storage.setItem(PushRegistration.TOKEN_KEY, deviceToken);
+
           if (isCordovaAndroid()) {
             this.subscribeToFirebaseTopic(variantId);
             for (const category of categories) {
               this.subscribeToFirebaseTopic(category);
             }
           }
+
+          resolve();
+        }
+      )
+      .catch(reject);
+    });
+  }
+
+  /**
+   * Unregister device for Android or IOS platforms
+   */
+  public unregister(): Promise<void> {
+    if (!window || !window.device) {
+      return Promise.reject(new Error("Registration requires cordova plugin. Verify the " +
+        "@aerogear/cordova-plugin-aerogear-push plugin is installed."));
+    }
+
+    if (!this.pushConfig || !this.pushConfig.config || !this.pushConfig.url) {
+      return Promise.reject(new Error("UPS registration: configuration is invalid"));
+    }
+
+    const storage = window.localStorage;
+    const deviceToken = storage.getItem(PushRegistration.TOKEN_KEY);
+
+    if (!deviceToken) {
+      return Promise.reject(new Error("Device token should not be empty"));
+    }
+
+    let platformConfig;
+    const url = this.pushConfig.url;
+    if (isCordovaAndroid()) {
+      platformConfig = this.pushConfig.config.android;
+    } else if (isCordovaIOS()) {
+      platformConfig = this.pushConfig.config.ios;
+    } else {
+      return Promise.reject(new Error("UPS registration: Platform is not supported."));
+    }
+
+    if (!platformConfig) {
+      return Promise.reject(new Error("UPS registration: Platform is configured." +
+        "Please add UPS variant and generate mobile - config.json again"));
+    }
+
+    const variantId = platformConfig.variantId || platformConfig.variantID;
+    if (!variantId) {
+      return Promise.reject(new Error("UPS registration: variantId is not defined."));
+    }
+
+    const variantSecret = platformConfig.variantSecret;
+    if (!variantSecret) {
+      return Promise.reject(new Error("UPS registration: variantSecret is not defined."));
+    }
+
+    this._objectInstance = window.PushNotification.init(
+      {
+        android: {},
+        ios: {
+          alert: true,
+          badge: true,
+          sound: true
+        }
+      }
+    );
+
+    const authToken = window.btoa(`${variantId}:${variantSecret}`);
+
+    const http = axios.create({
+      baseURL: url,
+      timeout: 5000,
+      headers: {"Authorization": `Basic ${authToken}`}
+    });
+
+    return new Promise((resolve, reject) => {
+      const endpoint = PushRegistration.API_PATH + "/" + deviceToken;
+      return http.delete(endpoint, {})
```

Comment:
Missing catch

Reply:
Hrm, there is weird. There is a [catch block there](https://github.com/aerogear/aerogear-js-sdk/blob/d6a0e573226090c7da37b22b0ebd4f2278a5168f/packages/push/src/PushRegistration.ts#L199).